File: FFVSDK/samples/CSharp/ChooseScannerForm.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

using Neurotec.Biometrics;

namespace CSharpSample
{
	public partial class ChooseScannerForm : Form
	{
		private string _allModuleString = string.Empty;
		private string _scannerString = string.Empty;

		public ChooseScannerForm()
		{
			InitializeComponent();

			_allModuleString = Nffv.GetAvailableScannerModules();
			if (_allModuleString.Length > 0)
			{
				string[] allModuleList = _allModuleString.Split(new char[] { ';' });
				clbScannerList.Items.Clear();
				clbScannerList.Items.AddRange(allModuleList);
			}
			else
			{
				clbScannerList.Items.Add("No scanner modules found.");
				clbScannerList.Enabled = false;
			}
		}

		private void btnOK_Click(object sender, EventArgs e)
		{
			if (_allModuleString.Length > 0)
			{
				StringBuilder scannerString = new StringBuilder();
				foreach (string selItem in clbScannerList.CheckedItems)
				{
					if (scannerString.Length > 0)
					{
						scannerString.Append(";");
					}
					scannerString.Append(selItem);
				}
				_scannerString = scannerString.ToString();
				Close();
			}
		}

		public string ScannerString
		{
			get
			{
				return _scannerString;
			}
		}

		public string FingerprintDatabase
		{
			get
			{
				return tbDatabaseName.Text;
			}
			set
			{
				tbDatabaseName.Text = value;
			}
		}

		public string FingerprintDatabasePassword
		{
			get
			{
				return tbPassword.Text;
			}
			set
			{
				tbPassword.Text = value;
			}
		}

		public string UserDatabase
		{
			get
			{
				return tbUserDatabase.Text;
			}
			set
			{
				tbUserDatabase.Text = value;
			}
		}
	}
}

File: FFVSDK/samples/VB.NET/ChooseScannerForm.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms

Imports Neurotec.Biometrics

Namespace VBNETSample
	Partial Public Class ChooseScannerForm
		Inherits Form
		Private _allModuleString As String = String.Empty
		Private _scannerString As String = String.Empty

		Public Sub New()
			InitializeComponent()

			_allModuleString = Nffv.GetAvailableScannerModules()
			If _allModuleString.Length > 0 Then
				Dim allModuleList As String() = _allModuleString.Split(New Char() {";"c})
				clbScannerList.Items.Clear()
				clbScannerList.Items.AddRange(allModuleList)
			Else
				clbScannerList.Items.Add("No scanner modules found.")
				clbScannerList.Enabled = False
			End If
		End Sub

		Private Sub btnOK_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnOK.Click
			If _allModuleString.Length > 0 Then
				Dim scannerString As New StringBuilder()
				For Each selItem As String In clbScannerList.CheckedItems
					If scannerString.Length > 0 Then
						scannerString.Append(";")
					End If
					scannerString.Append(selItem)
				Next
				_scannerString = scannerString.ToString()
				Close()
			End If
		End Sub

		Public ReadOnly Property ScannerString() As String
			Get
				Return _scannerString
			End Get
		End Property

		Public Property FingerprintDatabase() As String
			Get
				Return tbDatabaseName.Text
			End Get
			Set(ByVal value As String)
				tbDatabaseName.Text = Value
			End Set
		End Property

		Public Property FingerprintDatabasePassword() As String
			Get
				Return tbPassword.Text
			End Get
			Set(ByVal value As String)
				tbPassword.Text = Value
			End Set
		End Property

		Public Property UserDatabase() As String
			Get
				Return tbUserDatabase.Text
			End Get
			Set(ByVal value As String)
				tbUserDatabase.Text = Value
			End Set
		End Property
	End Class
End Namespace
